### PR TITLE
WW-4634  Centre alignment doesn't seem to work in Velocity tags

### PR DESCRIPTION
**Pull request by victorsosa**
Fix for     WW-4634

Centre alignment doesn't seem to work in Velocity tags

### CODE REVIEW

File: core/src/main/resources/template/xhtml/controlheader.ftl [hunk 1]
```diff
@@ -21,5 +21,8 @@
  */
 -->
 <#include "/${parameters.templateDir}/${parameters.expandTheme}/controlheader-core.ftl" />
-    <td
+    <td 
+        <#if parameters.align??>
+            class="tdAlign${parameters.align?html}"
```
Comment (lukaszlenart) [inline core/src/main/resources/template/xhtml/controlheader.ftl]: Can you use `-` to separate prefix and suffix?

**Comment by lukaszlenart** [comment]
This doesn't fix the main issue, right?

File: core/src/main/resources/template/xhtml/controlheader.ftl (hunk 1 above)
Comment (victorsosa) [inline core/src/main/resources/template/xhtml/controlheader.ftl]: You mean like this class="tdAlign-${parameters.align?html}" ?

Comment (lukaszlenart) [inline core/src/main/resources/template/xhtml/controlheader.ftl]: yes

**Comment by victorsosa** [comment]
Yes, it fix the main issue and keep the back compatibility.

File: core/src/main/resources/template/xhtml/controlheader.ftl (hunk 1 above)
Comment (victorsosa) [inline core/src/main/resources/template/xhtml/controlheader.ftl]: Done

**Comment by victorsosa** [comment]
Another thing I will change the name to just 

> align-center

because this css class can be used in other places that are using the align attribute; which is s Not supported in HTML5.

**Comment by aleksandr-m** [comment]
Why not just add some CSS class to this table cell? Then end-users can style it as they want and align parameter will not be needed.

**Comment by victorsosa** [comment]
CSS class to this table cell is what we have now and to have back compatibility as @lukaszlenart said.

**Comment by lukaszlenart** [comment]
I'm just wondering what was the main reason, I have prepared a unit test and it didn't pass when using 2.3.29-SNAPSHOT which means 2.3.28.1 is broken the same way.

**Comment by victorsosa** [comment]
@lukaszlenart  the parameter align was remove on this commit 
https://github.com/victorsosa/struts/commit/a0b34806b0f700a1fd240b698c4cec474711bbd0

**Comment by aleksandr-m** [comment]
Issue is about Velocity tags, not FreeMarker. Are we still supporting Velocity tags?

**Comment by lukaszlenart** [comment]
So it's not a regression bug ;-) Yes, we do support Velocity.

**Comment by aleksandr-m** [comment]
How the Velocity templating is working? The only Velocity templates I see are in the [archive](https://github.com/apache/struts/tree/master/core/src/main/resources/template/archive).

**Comment by aleksandr-m** [comment]
Yep, the aforementioned commit misses css class for that cell in the `controlheader.ftl`. It should be the same way as for the buttons. :)

**Comment by victorsosa** [comment]
OH ok, got you https://github.com/apache/struts/tree/master/core/src/main/resources/template/archive I miss that.

**Comment by victorsosa** [comment]
the css class for that cell is in the controlheader.ftl already.

**Comment by lukaszlenart** [comment]
All the tags are implemented in FreeMarker, those archive templates are deprecated and should be thrown away.

**Comment by aleksandr-m** [comment]
@lukaszlenart Ahh ok. Thanks for clarifying that.

**Comment by aleksandr-m** [comment]
@victorsosa I mean something like `<td class="tdInput">` in controlheader.ftl and `.tdInput {text-align:left;}` in styles.css. <-- This is for 2.5.

And if alignment doesn't work in 2.3 and your solution fixes it, then this PR should target `support-2-3` branch.

**Comment by victorsosa** [comment]
It will break back compatibility as @lukaszlenart said.

**Comment by aleksandr-m** [comment]
@victorsosa You see, this commit already *broke* backward compatibility, but it was incomplete in the way it didn't introduce css class for the input cell.
It is 2.5 so some breaking changes are expected.

**Comment by lukaszlenart** [comment]
@victorsosa this commit a0b34806b0f700a1fd240b698c4cec474711bbd0 dropped `align` attribute for buttons, it shouldn't affect `textfield` tag. Anyway, I think we should merge this PR and if needed prepare another one to target 2.3

**Comment by victorsosa** [comment]
@lukaszlenart ok

**Comment by aleksandr-m** [comment]
I would prefer to follow html5 and drop align attribute completely from S2 tags, BUT @victorsosa can you at least add some css class to input cell along with your solution.
Something like:
```
if align then class="tdInput align-${parameters.align?html}" else class="tdInput"
```

**Comment by victorsosa** [comment]
@aleksandr-m  You mean something like <td class="tdInput"> in controlheader.ftl and .tdInput {text-align:left;} in styles.css. <-- This is for 2.5.

THat means **drop align attribute completely** from S2 tags; So from now on the only way to set the align to _center_ for example is to overwrite the css class _tdInput_ in 2.5, right?

I will change to that code.

**Comment by aleksandr-m** [comment]
@victorsosa IMO it would be great. BUT if we really really want it to be backward compatible then we can do both. Add a class for future **and** use align attribute with `class="align-${parameters.align?html}"` like you propose in this PR.

**Comment by cnenning** [comment]
:+1:

**Comment by lukaszlenart** [comment]
Tests are failing :(
```
Error Message

expected:<...bel:</label></td><td[]><labelid="myname"ti...> but was:<...bel:</label></td><td[class="tdInput"]><labelid="myname"ti...>
Stacktrace

junit.framework.ComparisonFailure: expected:<...bel:</label></td><td[]><labelid="myname"ti...> but was:<...bel:</label></td><td[class="tdInput"]><labelid="myname"ti...>
	at junit.framework.Assert.assertEquals(Assert.java:100)
	at junit.framework.Assert.assertEquals(Assert.java:107)
	at junit.framework.TestCase.assertEquals(TestCase.java:269)
	at org.apache.struts2.views.jsp.AbstractUITagTest.verify(AbstractUITagTest.java:246)
	at org.apache.struts2.views.jsp.ui.LabelTest.testSimple(LabelTest.java:49)
```

**Comment by lukaszlenart** [comment]
103 tests need to be updated, I think it isn't worth

**Comment by victorsosa** [comment]
I will fix the 103 test cases, give me some time

**Comment by lukaszlenart** [comment]
@victorsosa osm!

**Comment by lukaszlenart** [comment]
I assume we're ok with this and this PR can be merged?

**Comment by victorsosa** [comment]
Yes, I am ok

**Comment by aleksandr-m** [comment]
Looks fine. Good job.

**Pull request merged**